static/slider.js

```diff
@@ -90,56 +111,99 @@
                                 observationsFiltered[construct][0][sublevel]
                                     .concat(filtered);
                         }
-        
+
                         courseObservations[construct][course][sublevel] = filtered;
+                        horizontalCourse[construct][course][sublevel] = filtered;
                         allStars[construct] += filtered.length;
                     }
                 } else {
                     observationsFiltered[construct][course] = {};
+                    horizontalCourse[construct][course] = {};
 
                     for (var sublevel in observations[construct][course]) {
                         const filtered = observations[construct][course][sublevel]
                             .filter(date => date <= timestamp);
 
                         observationsFiltered[construct][course][sublevel] = filtered;
+                        horizontalCourse[construct][course][sublevel] = filtered;
                         allStars[construct] += observationsFiltered[construct][course][sublevel].length;
                     }
                 }
             }
         }
 
-        // Swap verticalStarChart content if any of construct was vertically merged.
+        // Swap verticalStarChart and horizontalStarChart content if any of construct was merged.
         swapVerticalStarChart(courseObservations, allStars);
+        swapHorizontalStarChart(horizontalCourse, allStars);
 
         // Change elements in table.
-        changeElements(observationsFiltered, allStars);
+        changeElements(observationsFiltered, allStars, joinMergedSublevels(observationsFiltered));
     }
 
     /**
      * Changes colored cells and those with observations count after filtering observations by date.
      * @param {Object} observationsFiltered - Observations filtered by date.
      * @param {Object} allStars - Observations count per construct.
      */
-    function changeElements(observationsFiltered, allStars) {
+    function changeElements(observationsFiltered, allStars, levels) {
```

Comment:
This method changes color, am I right? If so `changeElements` isn't a good name.